File: src/CSP-AF/CSPAgents/CSPUtilityServerGroups/CSPInstallAgentBootstrap.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Models;
using CloudSuitePortal.Services;
using CloudSuitePortal.Controllers;

namespace actions.CSP.CSPAgents.CSPUtilityServerGroups
{
    class CSPInstallAgentBootstrap
    {
        public void run(Dictionary<string, object> Params)
        {
            var service = new CSPAgentService();
            service.Install();
        }
    }
}

File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPInstallAgentBootstrap.java
package CSPAF.CSPAgents.CSPUtilityServerGroups;

import java.io.*;
import java.util.*;
import java.util.concurrent.TimeoutException;

import com.infor.cloudsuite.portal.singletenant.agents.services.*;

class CSPInstallAgentBootstrap
{
    public void run(HashMap<String, Object> params) throws IOException, InterruptedException, TimeoutException
    {
        CSPAgentServiceController agentServiceController = null;
        
        if (System.getProperty("os.name").contains("Windows")) {
           	agentServiceController = new CSPWindowsAgentServiceController();
        }
        else if (System.getProperty("os.name").contains("Linux")) {
            agentServiceController = new CSPLinuxAgentServiceController();
        }
        
        agentServiceController.install();
    }
}